### PR TITLE
RFC: Schema parser

### LINKED ISSUE #22: Schema
Great package!  Allows me to parse queries very quickly and easily!

Will schema parsing be added to this repo (written in c or cpp)?  I can see graphql.js has a schema parser here: http://graphql.org/docs/api-reference-schema-language/

My motivation is to do something similar to https://github.com/matthewmueller/graph.ql but with another language.

Thank you,
Barret

### PR DESCRIPTION
**Pull request by swolchok**
This adds support for the RFC schema language additions in https://github.com/facebook/graphql/pull/90 .

- [x] AST for schema language
- [x] Parsing for schema language
- [x] API for parsing w/schema language enabled
- [x] Simple tests
- [x] JsonVisitor support for schema language
- [x] Test schema-kitchen-sink.graphql in automation (after adding kitchen-sink.graphql automated test on master)

Should fix #22 when complete.

### CODE REVIEW

**Comment by swolchok** [comment]
Planning to hold off merging this until folks get a chance to return from the holidays.

File: GraphQLParser.cpp [hunk 1]
```diff
@@ -20,36 +20,56 @@ namespace graphql {
...
 
-  auto result = doParse(error, scanner);
+  auto result = doParse(error, scanner, enableSchema);
   yylex_destroy(scanner);
   return result;
 }
 
-std::unique_ptr<ast::Node> parseFile(FILE *file, const char **error) {
+std::unique_ptr<ast::Node> parseString(const char *text, const char **error) {
+  return parseStringImpl(text, error, false);
```
Comment (ttsugriy) [inline GraphQLParser.cpp]: nit: it's hard to guess what is the meaning of the boolean without reading a type signature of parseStringImpl, so it's usually a good practice to leave a comment `/* enableSchema */` right next to a parameter.

File: lexer.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#line 1 "lexer.cpp"
+#line 1 "/Users/swolchok/libgraphqlparser/lexer.cpp"
```
Comment (ttsugriy) [inline lexer.cpp]: it's probably not something you'd like to leave :)

Comment (swolchok) [inline lexer.cpp]: Ugh, this keeps cropping up. cmake seems to run the tools by passing their absolute path. thanks.

**Pull request merged**